### PR TITLE
Try fix oxidation

### PR DESCRIPTION
**Pull request by TimeEngineer**
# What has changed ?

- [ ] Does this PR affect the User ?

### CODE REVIEW

File: src/runtime.rs [hunk 1]
```diff
@@ -26,30 +26,37 @@ struct Stuff {
...
-            trio_current_trio_token_fn: trio_lowlevel.getattr("current_trio_token")?.into_py(py),
-            trio_reschedule_fn: trio_lowlevel.getattr("reschedule")?.into_py(py),
-            trio_current_task_fn: trio_lowlevel.getattr("current_task")?.into_py(py),
-            trio_wait_task_rescheduled: trio_lowlevel.getattr("wait_task_rescheduled")?.into_py(py),
+fn get_stuff(py: Python<'_>) -> &Stuff {
+    STUFF.get_or_init(py, || {
+        let trio_lowlevel = py.import("trio").unwrap().getattr("lowlevel").unwrap();
+        let outcome = py.import("outcome").unwrap();
+        Stuff {
+            // It seems dangerous to use multithreaded tokio runtime where each thread call the python GIL
```
Comment (FirelightFlagboy) [inline src/runtime.rs:36]: 1. Why ? The GIL isn't multi thread-safe ?
2. If find this comment need more context.

File: src/runtime.rs [hunk 2]
```diff
@@ -26,30 +26,37 @@ struct Stuff {
...
+            tokio_runtime: tokio::runtime::Builder::new_current_thread()
+                .build()
+                .unwrap(),
```
Comment (FirelightFlagboy) [inline src/runtime.rs:39]: The previous implementation used a multi-thread runtime, could you measure the performance impact of this change ?

**Comment by TimeEngineer** [comment]
I found this where he says he has a deadlock and this is his fix commit: https://gitlab.com/ThibaultLemaire/pyo3-futures-sandox/-/commit/055a75b5eea349c7aaf75d30483da0e3c65c0928

**Comment by FirelightFlagboy** [comment]
I see, the potential deadlock would be 

```rust
Python::with_gil(|py| {
    // ..
    Python::with_gil(/* .. */); // Deadlock here
    // ..
})
```

**Pull request closed without merging**